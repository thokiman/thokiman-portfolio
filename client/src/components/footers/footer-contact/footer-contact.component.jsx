import useWindowDimensions from 'components/hooks/window-dimensions/useWindowDimensions.component';
import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectIsSideBarActive } from '../../../redux/header/header.selectors';
import {
  measureFooterGeneralIconStyles,
  measureProfileFooterIconStyles,
} from '../commons/footer.utils.style';
import {
  FooterContainerContact,
  FooterCreamPrickInklaim,
  FooterGeneralIcon,
  FooterWhiteOutlineThokiman,
  ProfileFooterContainerContact,
  ProfileFooterIconContact,
  ProfileFooterTextContact,
} from './footer-contact.styles';

export const FooterContact = ({ isSideBarActive }) => {
  const { viewWidth, viewHeight } = useWindowDimensions();
  return (
    <FooterContainerContact
      $issidebaractive={isSideBarActive}
      className='footer-contact'
    >
      <ProfileFooterContainerContact
        $measureprofilefootericon={measureProfileFooterIconStyles(
          viewWidth,
          viewHeight
        )}
      >
        <ProfileFooterIconContact>
          <FooterGeneralIcon
            $measurefootergeneralicon={measureFooterGeneralIconStyles(
              viewWidth,
              viewHeight
            )}
            className='footer-general-icon'
          >
            <FooterCreamPrickInklaim />
          </FooterGeneralIcon>
          <FooterGeneralIcon
            $measurefootergeneralicon={measureFooterGeneralIconStyles(
              viewWidth,
              viewHeight
            )}
            className='footer-general-icon'
          >
            <FooterWhiteOutlineThokiman />
          </FooterGeneralIcon>
        </ProfileFooterIconContact>
        <ProfileFooterTextContact className='footer-text'>
          Copyright<sup>&#9400;</sup>2021.
          <em>Allright reserved </em>
          {window.innerWidth <= 800 ? null : '|'}
          <br />
          Thomas Santosa, B.Eng.
        </ProfileFooterTextContact>
      </ProfileFooterContainerContact>
      <br />
    </FooterContainerContact>
  );
};
const mapStateToProps = createStructuredSelector({
  isSideBarActive: selectIsSideBarActive,
});

export default connect(mapStateToProps)(FooterContact);
