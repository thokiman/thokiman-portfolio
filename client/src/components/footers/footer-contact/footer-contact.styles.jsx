import styled, { css, keyframes } from 'styled-components';
import { ReactComponent as CreamPrickInklaim } from '../../../assets/collections-icon/prick-inklaim-icon/logo/prick_inklaim_logo_cream.svg';
import { ReactComponent as WhiteOutlineThokiman } from '../../../assets/collections-icon/thokiman-icon/logo/outline/thokiman_head_logo_cream.svg';

const blackColor = '#000000';
const heavyLightBlueColor = '#bccfd5';
const whiteSmokeColor = '#e3e1e4';
const maxWidthSmartphone = 800;

const FooterIsActivated = css`
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    position: relative;
    height: 16vh;
    width: 100vw;
    margin-bottom: 20px;
    left: 0;
    transition: left 0.5s ease;
  }
`;

const getFooterSideBarActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return null;
  return FooterIsActivated;
};
export const FooterContainerContact = styled.div`
  background-color: ${blackColor};
  height: 16vh;
  display: flex;
  align-items: center;
  justify-content: center;

  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    position: relative;
    top: 0;
    left: 140px;
    right: 0;
    bottom: 0;
    transition: left 0.5s ease;
    width: 64vw;
  }
  ${getFooterSideBarActive}
`;

export const ProfileFooterContainerContact = styled.div`
  margin-right: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 60vw;
`;

const piCreamAnimate = keyframes`
                from {
                    stroke: ${heavyLightBlueColor};
                    opacity: 0.6;
                }

                to {
                    stroke: ${whiteSmokeColor};
                    opacity: 1;
                }

`;

export const ProfileFooterIconContact = styled.div`
  width: 200px;
  height: 8vh;
  display: flex;
  justify-content: center;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 130px;
  }
  @media only screen and (max-width: 370px) {
    width: 100px;
  }
  .thoki-out-cream {
    animation: ${piCreamAnimate} 1s ease-in-out infinite;
  }

  .pi-cream {
    animation: ${piCreamAnimate} 1s ease-in-out infinite;
  }

  .pi-cream-one {
    animation: ${piCreamAnimate} 1s ease-in-out infinite;
  }
`;

const FooterIconContact = css`
  width: 100%;
  height: 100%;
  margin: 2px;
`;

export const FooterCreamPrickInklaim = styled(CreamPrickInklaim)`
  ${FooterIconContact}
`;
export const FooterWhiteOutlineThokiman = styled(WhiteOutlineThokiman)`
  ${FooterIconContact}
`;

export const ProfileFooterTextContact = styled.div`
  width: 100%;
  font-family: futura-medium;
  font-kerning: auto;
  letter-spacing: 0.03em;
  text-align: center;
  color: ${whiteSmokeColor};
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 15px;
  }
  @media only screen and (max-width: 370px) {
    font-size: 10px;
  }
`;

FooterContainerContact.displayName = 'FooterContainerContact';
ProfileFooterContainerContact.displayName = 'ProfileFooterContainerContact';
ProfileFooterIconContact.displayName = 'ProfileFooterIconContact';
FooterCreamPrickInklaim.displayName = 'FooterCreamPrickInklaim';
FooterWhiteOutlineThokiman.displayName = 'FooterWhiteOutlineThokiman';
ProfileFooterTextContact.displayName = 'ProfileFooterTextContact';
