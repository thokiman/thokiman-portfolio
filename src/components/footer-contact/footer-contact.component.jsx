import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectIsSideBarHidden } from "../../redux/header/header.selectors";

import {
  FooterContainerContact,
  FooterCreamPrickInklaim,
  FooterWhiteOutlineThokiman,
  ProfileFooterContainerContact,
  ProfileFooterIconContact,
  ProfileFooterTextContact,
} from "./footer-contact.styles";

const FooterContact = ({ isSideBarHidden }) => {
  return (
    <FooterContainerContact $issidebarhidden={isSideBarHidden}>
      <ProfileFooterContainerContact>
        <ProfileFooterIconContact>
          <FooterCreamPrickInklaim />
          <FooterWhiteOutlineThokiman />
        </ProfileFooterIconContact>
        <ProfileFooterTextContact>
          Copyright<sup>&#9400;</sup>2021.
          <em>Allright reserved </em>| Thomas Santosa, B. Eng.
        </ProfileFooterTextContact>
      </ProfileFooterContainerContact>
      <br />
    </FooterContainerContact>
  );
};
const mapStateToProps = createStructuredSelector({
  isSideBarHidden: selectIsSideBarHidden,
});

export default connect(mapStateToProps)(FooterContact);
