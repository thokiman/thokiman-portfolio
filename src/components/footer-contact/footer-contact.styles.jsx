import styled, { keyframes, css } from "styled-components";

import { ReactComponent as CreamPrickInklaim } from "../../assets/collections-icon/prick-inklaim-icon/logo/prick_inklaim_logo_cream.svg";
import { ReactComponent as WhiteOutlineThokiman } from "../../assets/collections-icon/thokiman-icon/logo/outline/thokiman_head_logo_cream.svg";

const FooterSideBarActive = css`
  @media only screen and (max-width: 800px) {
    position: relative;
    height: 16vh;
    width: 100vw;
    margin-bottom: 20px;
    left: 0;
    transition: left 0.5s ease;
  }
`;
const getFooterSideBarActive = (props) => {
  if (props.$issidebarhidden) return null;
  return FooterSideBarActive;
};
export const FooterContainerContact = styled.div`
  background-color: #3f4249;
  height: 16vh;
  display: flex;
  align-items: center;
  justify-content: center;

  @media only screen and (max-width: 800px) {
    position: relative;
    top: 0;
    left: 140px;
    right: 0;
    bottom: 0;
    transition: left 0.5s ease;
    width: 66vw;
  }
`;

export const ProfileFooterContainerContact = styled.div`
  margin-right: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 60vw;
`;

const piCreamAnimate = keyframes`
                from {
                    stroke: #bccfd5;
                    opacity: 0.6;
                }

                to {
                    stroke: #e3e1e4;
                    opacity: 1;
                }

`;

export const ProfileFooterIconContact = styled.div`
  width: 200px;
  height: 8vh;
  display: flex;
  justify-content: center;
  @media only screen and (max-width: 800px) {
    width: 150px;
  }
  .thoki-out-cream {
    animation: ${piCreamAnimate} 1s ease-in-out infinite;
  }

  .pi-cream {
    animation: ${piCreamAnimate} 1s ease-in-out infinite;
  }

  .pi-cream-one {
    animation: ${piCreamAnimate} 1s ease-in-out infinite;
  }
`;

const FooterIconContact = css`
  width: 100%;
  height: 100%;
  margin: 2px;
`;

export const FooterCreamPrickInklaim = styled(CreamPrickInklaim)`
  ${FooterIconContact}
`;
export const FooterWhiteOutlineThokiman = styled(WhiteOutlineThokiman)`
  ${FooterIconContact}
`;

export const ProfileFooterTextContact = styled.div`
  width: 100%;
  font-family: futura-medium;
  font-kerning: auto;
  letter-spacing: 0.03em;
  text-align: center;
  color: #e3e1e4;
  @media only screen and (max-width: 800px) {
    font-size: 15px;
  }
`;

FooterContainerContact.displayName = "FooterContainerContact";
ProfileFooterContainerContact.displayName = "ProfileFooterContainerContact";
ProfileFooterIconContact.displayName = "ProfileFooterIconContact";
FooterCreamPrickInklaim.displayName = "FooterCreamPrickInklaim";
FooterWhiteOutlineThokiman.displayName = "FooterWhiteOutlineThokiman";
ProfileFooterTextContact.displayName = "ProfileFooterTextContact";
