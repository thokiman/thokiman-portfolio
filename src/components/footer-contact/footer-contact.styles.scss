.footer-container-contact {
    position: relative;
    top: 10px;
    bottom: 10px;
    height: 14vh;
    display: flex;
    justify-content: center;

    .profile-footer-container-contact {

        position: relative;
        right: 30px;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        width: 39%;
        height: 100%;

        .profile-footer-icon-contact {
            display: flex;
            flex-direction: row;
            justify-content: center;
            align-items: center;
            width: 30%;
            height: 65%;

            .thoki-out-cream {
                animation: pi-cream-animate 1s ease-in-out infinite;
            }

            .pi-cream {
                animation: pi-cream-animate 1s ease-in-out infinite;

            }

            .pi-cream-one {
                animation: pi-cream-animate 1s ease-in-out infinite;

            }

            @keyframes pi-cream-animate {
                from {
                    stroke: #bccfd5;
                    opacity: 0.6;
                }

                to {
                    stroke: #e3e1e4;
                    opacity: 1;
                }
            }

            .footer-icon-contact {
                width: 100%;
                height: 100%;

            }
        }

        .profile-footer-text-contact {

            width: 100%;
            font-family: futura-medium;
            font-kerning: auto;
            letter-spacing: 0.03em;
            text-align: center;
            color: #e3e1e4;
        }
    }
}