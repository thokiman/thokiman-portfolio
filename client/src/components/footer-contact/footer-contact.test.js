import React from "react";
import { shallow } from "enzyme";

import { FooterContact } from "./footer-contact.component";

describe("FooterContact component", () => {
  let wrapper;
  const mockFalseBool = false;

  beforeEach(() => {
    const mockProps = {
      isSideBarHidden: mockFalseBool,
    };
    wrapper = shallow(<FooterContact {...mockProps} />);
  });

  it("should render FooterContact component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
