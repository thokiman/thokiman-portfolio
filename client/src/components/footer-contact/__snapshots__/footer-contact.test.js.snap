// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterContact component should render FooterContact component 1`] = `ShallowWrapper {}`;
